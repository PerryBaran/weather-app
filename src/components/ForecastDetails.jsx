import React from "react";
import "../styles/ForecastDetails.css";
import PropTypes from "prop-types";

function ForecastDetails({ forecast }) {
  const {
    date,
    humidity,
    temperature: { max, min },
    wind: { speed, direction },
  } = forecast;

  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__max-temp">
        <span>Max Temperature:</span> {max}&deg;C
      </div>
      <div className="forecast-details__min-temp">
        <span>Min Temperature:</span> {min}&deg;C
      </div>
      <div className="forecast-details__humidity">
        <span>Humidity:</span> {humidity}%
      </div>
      <div className="forecast-details__wind">
        <span>Wind:</span> {speed}mph {direction}
      </div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }),
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};

export default ForecastDetails;
