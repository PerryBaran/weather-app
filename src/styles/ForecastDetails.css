.forecast-details {
  margin-top: 20px;
  text-align: center;
}

.forecast-details span {
  font-weight: 600;
}