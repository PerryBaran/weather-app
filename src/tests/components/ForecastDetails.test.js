import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    forecast: {
      date: 12345,
      humidity: 80,
      temperature: {
        max: 20,
        min: 10,
      },
      wind: {
        speed: 40,
        direction: "ne",
      },
    },
  };

  describe("snapshot", () => {
    it("renders correctly", () => {
      const { asFragment } = render(
        <ForecastDetails forecast={validProps.forecast} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("props", () => {
    it("renders correct values for props", () => {
      const { getByText } = render(
        <ForecastDetails forecast={validProps.forecast} />
      );

      expect(
        getByText(new Date(validProps.forecast.date).toDateString())
      ).toHaveAttribute("class", "forecast-details__date");
      expect(
        getByText(`${validProps.forecast.temperature.max}°C`)
      ).toHaveAttribute("class", "forecast-details__max-temp");
      expect(
        getByText(`${validProps.forecast.temperature.min}°C`)
      ).toHaveAttribute("class", "forecast-details__min-temp");
      expect(getByText(`${validProps.forecast.humidity}%`)).toHaveAttribute(
        "class",
        "forecast-details__humidity"
      );
      expect(
        getByText(
          `${validProps.forecast.wind.speed}mph ${validProps.forecast.wind.direction}`
        )
      ).toHaveAttribute("class", "forecast-details__wind");
    });
  });
});
