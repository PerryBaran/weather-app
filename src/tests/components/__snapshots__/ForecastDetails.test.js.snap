// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__max-temp"
    >
      <span>
        Max Temperature:
      </span>
       20°C
    </div>
    <div
      class="forecast-details__min-temp"
    >
      <span>
        Min Temperature:
      </span>
       10°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      <span>
        Humidity:
      </span>
       80%
    </div>
    <div
      class="forecast-details__wind"
    >
      <span>
        Wind:
      </span>
       40mph ne
    </div>
  </div>
</DocumentFragment>
`;
